### DISCUSSION TITLE
BG generator fo xDrip

### DISCUSSION THREAD

**Discussion by aptash** [NightscoutFoundation/xDrip#3277] [General]
Hello!

Is there any BG generators for xDrip for development purpose?

Currently I use random BG from aaps to xDrip.
But the thing is I need to test sending from xDrip to aaps.

Ideally it would be to use Eversense as the source.

Thanks in advance!

**Comment by Navid200** [comment]
xDrip has a fake data source for test purposes.
https://navid200.github.io/xDrip/docs/FakeDataSource.html

**Reply by aptash** [reply]
Thanks a lot for the quick response!
I managed to use it.

But I need something to test a bug with backfilled data. 
Something to emulate Bluetooth connection problem that restored after a few minutes. So xDrip receives all missed data, not only the last BG.

I presume it's not possible with "Fake/Mock Data Source".
Or can I do it?
